### ISSUE TITLE
Linformer

### ISSUE THREAD

**Issue by Laksh1997** [huggingface/transformers#5201]
# 🌟 New model addition

## Model description

https://arxiv.org/pdf/2006.04768.pdf

This model is very simple, it just projects the key tensor into a lower dimensional space (e.g k=128) along the sequence axis, then computes attention (seq_len x k), softmax, and matmul with the value tensor (note, the value tensor must also be projected to dimension k along the length dimension).

Could this be added?

## Open source status

Pytorch sketch implementation: https://github.com/tatp22/linformer-pytorch

**Comment by DNGros** [comment]
Duplicate #4967
